### PR TITLE
Simplify a few expressions

### PR DESCRIPTION
**Pull request by hukkinj1**

### CODE REVIEW

File: mutmut/__init__.py [hunk 1]
```diff
@@ -1028,7 +1028,7 @@ def kill(process_):
                     if not line:
                         break
                     callback(line)
-        except (IOError, OSError):
+        except OSError:
```
Comment (hukkinj1) [inline mutmut/__init__.py:1031]: See https://docs.python.org/3/library/exceptions.html#OSError

> Changed in version 3.3: EnvironmentError, IOError, WindowsError, socket.error, select.error and mmap.error have been merged into OSError, and the constructor may return a subclass.

File: mutmut/__init__.py [hunk 2]
```diff
@@ -1090,7 +1090,7 @@ def write(self, s):
 
     modules_to_force_unload = {x.partition(os.sep)[0].replace('.py', '') for x in config.paths_to_mutate}
 
-    for module_name in list(sorted(set(sys.modules.keys()) - set(modules_before), reverse=True)):
+    for module_name in sorted(set(sys.modules.keys()) - set(modules_before), reverse=True):
```
Comment (hukkinj1) [inline mutmut/__init__.py:1093]: `sorted()` returns a list, so no need to wrap in `list()`

File: mutmut/cache.py [hunk 3]
```diff
@@ -158,7 +158,7 @@ def print_result_cache(show_diffs=False, dict_synonyms=None, print_only_filename
     print('')
 
     def print_stuff(title, mutant_query):
-        mutant_list = list(sorted(mutant_query, key=lambda x: x.line.sourcefile.filename))
+        mutant_list = sorted(mutant_query, key=lambda x: x.line.sourcefile.filename)
```
Comment (hukkinj1) [inline mutmut/cache.py:161]: sorted() returns a list, so no need to wrap in list()

File: mutmut/__init__.py [hunk 4]
```diff
@@ -236,7 +236,7 @@ def number_mutation(value, **_):
 
 
 def string_mutation(value, **_):
-    prefix = value[:min([x for x in [value.find('"'), value.find("'")] if x != -1])]
+    prefix = value[:min(x for x in [value.find('"'), value.find("'")] if x != -1)]
```
Comment (hukkinj1) [inline mutmut/__init__.py:239]: min() can take a generator

File: mutmut/__init__.py (hunk 1 above)
Comment (boxed) [reply mutmut/__init__.py:1031]: Nice. I will try to remember that.

**Comment by boxed** [comment]
Thanks!

**Pull request merged**